man/systemd-networkd-wait-online.service.xml

```diff
@@ -144,6 +144,20 @@
         <xi:include href="version-info.xml" xpointer="v249"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--dns</option></term>
+
+        <listitem><para>Waiting for a DNS server of each network interface to be accessible. If this
+        option is specified with <option>--any</option>, then
+        <command>systemd-networkd-wait-online</command> exits with success when at least one interface
+        becomes online and has an accessible DNS server.</para>
+        <para>If there are search domains configured on a link, or if a link has <varname>DNSDefaultRoute=yes</varname>,
+        then there must be an accessible link-specific DNS server configured. Otherwise, global DNS servers
+        may be considered.</para>
```

Comment:
hmm, i think we should just look for interfaces where default route is on, or where the "." domain is listed as search domain, and ignore all interfaces whether default route is off and no "." search domain is listed.

Reply:
Can you elaborate a bit on your reasoning here? Why only consider the '.' domain? I was thinking that if any domains are configured on a link, then servers also need to be configured for those domains to be useful, no?

Reply:
Thanks for clarifying. I think the latest push addresses this better.